plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHiveRoles.java

```diff
@@ -0,0 +1,574 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import io.trino.Session;
+import io.trino.spi.security.Identity;
+import io.trino.spi.security.SelectedRole;
+import io.trino.spi.type.Type;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.MaterializedResult;
+import io.trino.testing.QueryRunner;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static io.trino.plugin.hive.HiveQueryRunner.HIVE_CATALOG;
+import static io.trino.spi.type.VarcharType.createUnboundedVarcharType;
+import static io.trino.testing.QueryAssertions.assertContains;
+import static io.trino.testing.QueryAssertions.assertEqualsIgnoreOrder;
+import static org.testng.Assert.assertEquals;
+
+@Test(singleThreaded = true)
+abstract class AbstractTestHiveRoles
+        extends AbstractTestQueryFramework
+{
+    private final boolean legacyCommands;
```

Comment:
I would use abstract a method instead. Passing dependencies in constructors usually does not work well in complex cases. So far it is not an issue, but it just a bad habit.

Reply:
In this case I think the constructor with a final boolean field is the right choice.  I understand there are problems if we are talking about more complex objects.